python/pyspark/sql/tests/test_pandas_udf_scalar.py

```diff
@@ -134,6 +134,30 @@ def test_pandas_udf_nested_arrays(self):
         result = df.select(tokenize("vals").alias("hi"))
         self.assertEqual([Row(hi=[["hi", "boo"]]), Row(hi=[["bye", "boo"]])], result.collect())
 
+    def test_pandas_array_struct(self):
+        # SPARK-38098: Support Array of Struct for Pandas UDFs and toPandas
+        # import numpy as np
+
+        @pandas_udf("Array<struct<col1:string, col2:long, col3:double>>")
+        def return_cols(cols):
+            # self.assertEqual(type(cols), pd.Series)
+            # self.assertEqual(type(cols[0]), np.ndarray)
+            # self.assertEqual(type(cols[0][0]), dict)
```

Comment:
I guess we can't use `self` in the udf. Shall we follow the other tests to use builtin `assert` instead:

https://github.com/apache/spark/blob/f8b3d5322e6cbce2e42a6940518686b7255e79cb/python/pyspark/sql/tests/test_pandas_udf_scalar.py#L1206-L1215

Also `import numpy as np` might need to be in the udf.

Reply:
Thank you @ueshin for the review and comments.
I have added the proposed modifications.
As for `import numpy as np`, I have now added it explicitly, however externally to the udf, just to be consistent with the other tests there that use numpy.